packages/sign_in_with_apple/lib/src/authorization_credential.dart

```diff
@@ -122,33 +109,33 @@ class AuthorizationCredentialPassword {
 
 // ignore_for_file: avoid_as, public_member_api_docs
 AuthorizationCredentialAppleID parseAuthorizationCredentialAppleID(
-  Map<dynamic, dynamic> response,
+  Map<dynamic, dynamic>? response,
```

Comment:
It might be helpful to require that `response` is not null: In the code below, `response == null` causes the `throw` expression to be evaluated, so the caller might as well ensure that it isn't null (it may be known at the call site).

The null-check in line 116 could then be eliminated (I also don't understand why it would be needed in that line and not in the following lines, cf. line 134), and the two invocations of `operator []` in line 114 and 125 could then omit the `?`.

A similar change could be useful with similar functions, including line 129